src/acpi-tables/src/fadt.rs

```diff
@@ -0,0 +1,149 @@
+/// fixed features.
+pub const FADT_F_HW_REDUCED_ACPI: u8 = 20;
+
+#[repr(packed)]
+#[derive(Debug, Copy, Clone, Default, AsBytes)]
+pub struct Fadt {
+    _header: SdtHeader,
+    _firmware_control: U32,
+    _dsdt: U32,
+    _reserved_1: u8,
+    _preferred_pm_profile: u8,
+    // In HW-reduced mode, fields starting from SCI_INT until CENTURY are ignored
+    _sci_int: U16,
+    _smi_cmd: U32,
+    _acpi_enable: u8,
+    _acpi_disable: u8,
+    _s4bios_req: u8,
+    _pstate_cnt: u8,
+    _pm1a_evt_blk: U32,
+    _pm1b_evt_blk: U32,
+    _pm1a_cnt_blk: U32,
+    _pm1b_cnt_blk: U32,
+    _pm2_cnt_blk: U32,
+    _pm_tmr_blk: U32,
+    _gpe0_blk: U32,
+    _gpe1_blk: U32,
+    _pm1_evt_len: u8,
+    _pm1_cnt_len: u8,
+    _pm2_cnt_len: u8,
+    _pm_tmr_len: u8,
+    _gpe0_blk_len: u8,
+    _gpe1_blk_len: u8,
+    _gpe1_base: u8,
+    _cst_cnt: u8,
+    _p_lvl2_lat: U16,
+    _p_lvl3_lat: U16,
+    _flush_size: U16,
+    _flush_stride: U16,
+    _duty_offset: u8,
+    _duty_width: u8,
+    _day_alrm: u8,
+    _mon_alrm: u8,
+    _century: u8,
+    _iapc_boot_arch: U16,
+    _reserved_2: u8,
+    _flags: U32,
+    _reset_reg: GenericAddressStructure,
+    _reset_value: u8,
+    _arm_boot_arch: U16,
+    _fadt_minor_version: u8,
+    _x_firmware_ctrl: U64,
+    _x_dsdt: U64,
+    // In HW-reduced mode, fields starting from X_PM1a_EVT_BLK through X_GPE1_BLK
+    // are ignored
+    _x_pm1a_evt_blk: GenericAddressStructure,
+    _x_pm1b_evt_blk: GenericAddressStructure,
+    _x_pm1a_cnt_blk: GenericAddressStructure,
+    _x_pm1b_cnt_blk: GenericAddressStructure,
+    _x_pm2_cnt_blk: GenericAddressStructure,
+    _x_pm_tmr_blk: GenericAddressStructure,
+    _x_gpe0_blk: GenericAddressStructure,
+    _x_gpe1_blk: GenericAddressStructure,
+    _sleep_control_reg: GenericAddressStructure,
+    _sleep_status_reg: GenericAddressStructure,
+    _hypervisor_vendor_id: [u8; 8],
+}
+
+impl Fadt {
+    pub fn new(oem_id: [u8; 6], oem_table_id: [u8; 8], oem_revision: u32) -> Self {
+        let header = SdtHeader::new(
+            *b"FACP",
+            // It's fine to unwrap here, we know that the size of the Fadt structure fits in 32
+            // bits.
+            std::mem::size_of::<Self>().try_into().unwrap(),
+            6, // revision 6
+            oem_id,
+            oem_table_id,
+            oem_revision,
+        );
+
+        Fadt {
+            _header: header,
+            _fadt_minor_version: 5,
+            ..Default::default()
+        }
+    }
+
+    /// Set the address of the DSDT table
+    ///
+    /// This sets the 64bit variant, X_DSDT field of the FADT table
+    pub fn set_x_dsdt(&mut self, addr: u64) {
+        self._x_dsdt = U64::new(addr);
+    }
+
+    /// Set the FADT flags
+    pub fn set_flags(&mut self, flags: u32) {
+        self._flags = U32::new(flags);
+    }
+
+    /// Set the IA-PC specific flags
+    pub fn setup_iapc_flags(&mut self, flags: u16) {
+        self._iapc_boot_arch = U16::new(flags);
```

Comment:
nit: no underscore on `iapc_boot_arch`

Reply:
I ignore the lint and everything is without an underscore now.